src/h3/mod.rs

```diff
@@ -2764,6 +2769,11 @@ mod tests {
 
         s.advance().ok();
 
+        // The server should consume the request to update flow control credit.
+        assert_eq!(s.server.poll(&mut s.pipe.server), Ok((0, ev_headers)));
+
+        s.advance().ok();
```

Comment:
It's not clear to me why these steps need to be added in the H3 unit tests, they don't seem to test any aspect of the flow control and Event::Headers already has pretty good coverage.

Reply:
Sorry I misread. So actually what is happening is that the flow control is only updated once the application reads stream data out of the library. In this case that happens when the app calls `poll()`

Reply:
Exactly. In this PR receiving packet into the receiver buffer will check the flow limit against MAX_DATA but doesn't update MAX_DATA (which make sense because updating MAX_DATA without the data consumed by the app only increases receiver buffer too much) Previously test was passed because receiving the packet will trigger MAX_DATA update.